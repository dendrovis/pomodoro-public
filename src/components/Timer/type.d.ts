import type {
  Chart,
  BubbleDataPoint,
  Point,
  _DeepPartialObject,
  ChartConfiguration,
} from 'chart.js';

export type Properties = {
  totalCycle: number;
  currentCycle: number;
  minutes: number;
  seconds: number;
  isDark?: boolean;
  isBreak: boolean;
  onPlay: () => void;
  onPause: () => void;
  onStop: () => void;
  onDone?: () => void;
  onBreak?: () => void;
};

export type ClassnamesIconPlayPause = (isDark?: boolean) => string;
export type ClassnamesIconStop = (isDark?: boolean) => string;

type TimerProps = { id: string } & Properties;

export type States = {
  [id: string]: Properties & { chartTimer?: Chart; chartCycle?: Chart; isPlay: boolean };
};

export type UpdateChart = (
  type:
    | { timer: { min: number; sec: number; isBreak: boolean } }
    | { cycle: { cycle: number; totalCycle: number; isBreak: boolean } },
  id: string
) => void;
export type UpdateChartInternal = (
  chartObjectID: 'chartTimer' | 'chartCycle',
  id: string,
  config: ChartConfiguration
) => void;

export type CreateArrayRepresentCycleCountInChart = (
  totalCycle: number
) => (number | [number, number] | Point | BubbleDataPoint | null)[]; // follow chart.js property acceptable type

export type CreateArrayRepresentCurrentCycleInChart = (
  totalCycle: number,
  currentCycle: number,
  isBreak: booleanm
) => (
  | string
  | _DeepPartialObject<CanvasGradient>
  | _DeepPartialObject<CanvasPattern>
  | undefined
)[]; // follow chart.js property acceptable type

export type CycleSpacing = (totalCycle: number) => number;
export type GetTimeInSec = (min: number, sec: number) => number;
export type GetPercentCurrentTimeInAnHour = (timeInSec: number, totalTime: number) => number;
export type GetDataChartTimerConfig = (
  id: string,
  min: number,
  sec: number,
  isBreak: boolean
) => ChartConfiguration;
export type GetDataChartCycleConfig = (
  id: string,
  totalCycle: number,
  currentCycle: number,
  isBreak: boolean
) => ChartConfiguration;

export type CreateChart = (
  elementID: string,
  id: string,
  config: ChartConfiguration,
  chartObjectID: 'chartTimer' | 'chartCycle'
) => void;

export type CreateClickEvent = (
  elementID: string,
  callback: EventListenerOrEventListenerObject
) => void;

export type InitializeComponent = () => void;

export type IsPlay = (id) => void;
export type OnPlay = (id) => void;
export type OnPause = (id) => void;
export type OnStop = (id) => void;

export type SetTheme = (id: string, value: 'dark' | 'light') => void;
export type SetCycle = (id: string, value: number) => void;
export interface TimerType {
  (props: TimerProps): string;
  setTheme: SetTheme;
  setCycle: SetCycle;
}
