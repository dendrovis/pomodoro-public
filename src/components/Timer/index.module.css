.container > div,
.timer > div,
.cycle > div {
  position: relative;
}

.cycle {
  position: absolute;
  top: 30px;
  left: 15px;

  & > div {
    width: min-content;
    filter: drop-shadow(0.3rem 0.3rem 0.1rem rgba(0 0 0 / 25%));
    min-width: 250px;
    max-width: 260px;
  }
}

.timer {
  filter: drop-shadow(0.3rem 0.3rem 0.1rem rgba(0 0 0 / 25%));
}

/* hide inconsistent hover effects due to element stack blocking */
.cover {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 304px;
  height: 304px;
}

.icon-play-pause {
  cursor: pointer;
  position: absolute;
  z-index: 2;
  top: 70px;
  left: 55px;
  filter: drop-shadow(0.3rem 0.3rem 0.1rem rgba(0 0 0 / 25%));

  & > svg {
    width: 180px;
    height: 180px;

    & > path {
      transition: all 0.3s ease-in-out;
      fill: var(--tertiary-color-dark, white);
    }
  }
}

.icon-stop {
  cursor: pointer;
  position: absolute;
  z-index: 2;
  bottom: 0;
  right: 0;
  filter: drop-shadow(0.3rem 0.3rem 0.1rem rgba(0 0 0 / 25%));

  & > svg {
    width: 50px;
    height: 50px;

    & > path {
      transition: all 0.3s ease-in-out;
      fill: var(--tertiary-color-dark, white);
    }
  }
}

.dark {
  & > svg > path {
    fill: var(--primary-color, white);
  }
}
