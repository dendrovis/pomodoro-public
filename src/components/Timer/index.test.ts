import { describe, expect, it, vi } from 'vitest';
import Timer from '.';

// currently most of the test here base on initial to intermediate behaviours before passing into chartjs apis
// TODO: to test end render is snapshot as images after chartjs rendered
describe('when given timer value in seconds', () => {
  it('should render with the given the right value', () => {
    expect(
      Timer({
        id: 'test',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toBeDefined();

    expect(
      Timer({
        id: 'test-2',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 60,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toBeDefined();
  });
  it('should throw error if value outside of boundary (0 to 60)', () => {
    expect(() =>
      Timer({
        id: 'test-3',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: -1,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toThrowError();

    expect(() =>
      Timer({
        id: 'test-4',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 61,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toThrowError();
  });
});

describe('when given timer value in minutes', () => {
  it('should render with the given the right value', () => {
    expect(
      Timer({
        id: 'test-5',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toBeDefined();

    expect(
      Timer({
        id: 'test-6',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 60,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toBeDefined();
  });
  it('should throw error if value outside of boundary (0 to 60)', () => {
    expect(() =>
      Timer({
        id: 'test-7',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: -1,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toThrowError();

    expect(() =>
      Timer({
        id: 'test-8',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 61,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toThrowError();
  });
});

describe('when given total cycle', () => {
  it('should render the number of cycle in the inner circle', () => {
    expect(
      Timer({
        id: 'test-9',
        totalCycle: 1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toBeDefined();

    expect(
      Timer({
        id: 'test-10',
        totalCycle: 48,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toBeDefined();
  });
  it('should throw error is 0 or negative value of cycles', () => {
    expect(() =>
      Timer({
        id: 'test-11',
        totalCycle: 0,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toThrowError();

    expect(() =>
      Timer({
        id: 'test-11',
        totalCycle: -1,
        currentCycle: 1,
        isBreak: false,
        minutes: 0,
        seconds: 0,
        onPlay: () => true,
        onPause: () => true,
        onStop: () => {},
      })
    ).toThrowError();
  });
});

describe('when clicked stop', () => {
  it('should trigger stop callbacks', () => {
    let called = false;
    const callback = () => {
      called = true;
    };

    //suppress side effect due to DOMContentLoaded not isolated called - chart error
    vi.spyOn(console, 'error').mockImplementation(() => undefined);

    document.body.innerHTML = Timer({
      id: 'test-12',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      minutes: 0,
      seconds: 0,
      onPlay: () => true,
      onPause: () => true,
      onStop: callback,
    });
    const componentID = '#icon-stop-test-12';
    document.dispatchEvent(new Event('DOMContentLoaded'));
    const element = document.querySelector(componentID);
    element && element.dispatchEvent(new Event('click'));
    expect(called).toBe(true);
  });
});

describe('when clicked play', () => {
  it('should trigger play callbacks', () => {
    let called = false;
    const callback = () => {
      called = true;
      return true;
    };

    //suppress side effect due to DOMContentLoaded not isolated called - chart error
    vi.spyOn(console, 'error').mockImplementation(() => undefined);

    document.body.innerHTML = Timer({
      id: 'test-13',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      minutes: 0,
      seconds: 0,
      onPlay: callback,
      onPause: () => true,
      onStop: () => {},
    });

    const componentID = '#icon-play-pause-test-13';
    document.dispatchEvent(new Event('DOMContentLoaded'));
    const element = document.querySelector(componentID);
    element && element.dispatchEvent(new Event('click'));
    expect(called).toBe(true);
  });
});

// must click play then can pause, since initial load always on play first, thus double click
describe('when clicked pause', () => {
  it('should trigger pause callbacks', () => {
    let called = false;
    const callback = () => {
      called = true;
      return true;
    };

    //suppress side effect due to DOMContentLoaded not isolated called - chart error
    vi.spyOn(console, 'error').mockImplementation(() => undefined);

    document.body.innerHTML = Timer({
      id: 'test-14',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      minutes: 0,
      seconds: 0,
      onPlay: () => true,
      onPause: callback,
      onStop: () => {},
    });

    const componentID = '#icon-play-pause-test-14';
    document.dispatchEvent(new Event('DOMContentLoaded'));
    const element = document.querySelector(componentID);
    if (element) {
      element.dispatchEvent(new Event('click'));
      element.dispatchEvent(new Event('click'));
    }
    expect(called).toBe(true);
  });
});

describe('when set theme', () => {
  it('should show dark theme when set dark', () => {
    const element = Timer({
      id: 'test-15',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      isDark: true,
      minutes: 0,
      seconds: 0,
      onPlay: () => true,
      onPause: () => true,
      onStop: () => {},
    });
    expect(element).toMatchSnapshot();

    document.body.innerHTML = Timer({
      id: 'test-16',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      isDark: false,
      minutes: 0,
      seconds: 0,
      onPlay: () => true,
      onPause: () => true,
      onStop: () => {},
    });
    Timer.setTheme('test-16', 'dark');
    expect(document.body.innerHTML).toMatchSnapshot();
  });
  it('should show light theme when set light', () => {
    const element = Timer({
      id: 'test-17',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      isDark: false,
      minutes: 0,
      seconds: 0,
      onPlay: () => true,
      onPause: () => true,
      onStop: () => {},
    });
    expect(element).toMatchSnapshot();

    document.body.innerHTML = Timer({
      id: 'test-18',
      totalCycle: 1,
      currentCycle: 1,
      isBreak: false,
      isDark: true,
      minutes: 0,
      seconds: 0,
      onPlay: () => true,
      onPause: () => true,
      onStop: () => {},
    });
    Timer.setTheme('test-18', 'light');
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
