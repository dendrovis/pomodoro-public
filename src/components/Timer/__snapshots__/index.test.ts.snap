// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when set theme > should show dark theme when set dark 1`] = `
"
    <div class="_container_f0165c">
      <div>
        
  <div class="_timer_f0165c">
    <div>
      <canvas id="timer-test-15"></canvas>
    </div>
  </div>

        <div class="_cover_f0165c"></div>
        
  <div class="_cycle_f0165c">
    <div>
      <canvas id="timer-cycle-test-15"></canvas>
    </div>
  </div>
        <div id="icon-play-pause-test-15" class="_icon-play-pause_f0165c _dark_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5 14.5L14.5 10L7.5 5.5V14.5ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"/>
</svg>
</div>
        <div id="icon-stop-test-15" class="_icon-stop_f0165c _dark_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M6 14H14V6H6V14ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"/>
</svg>
</div>
      </div>
    </div>
      "
`;

exports[`when set theme > should show dark theme when set dark 2`] = `
"
    <div class="_container_f0165c">
      <div>
        
  <div class="_timer_f0165c">
    <div>
      <canvas id="timer-test-16"></canvas>
    </div>
  </div>

        <div class="_cover_f0165c"></div>
        
  <div class="_cycle_f0165c">
    <div>
      <canvas id="timer-cycle-test-16"></canvas>
    </div>
  </div>
        <div id="icon-play-pause-test-16" class="_icon-play-pause_f0165c _dark_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5 14.5L14.5 10L7.5 5.5V14.5ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"></path>
</svg>
</div>
        <div id="icon-stop-test-16" class="_icon-stop_f0165c _dark_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M6 14H14V6H6V14ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"></path>
</svg>
</div>
      </div>
    </div>
      "
`;

exports[`when set theme > should show light theme when set light 1`] = `
"
    <div class="_container_f0165c">
      <div>
        
  <div class="_timer_f0165c">
    <div>
      <canvas id="timer-test-17"></canvas>
    </div>
  </div>

        <div class="_cover_f0165c"></div>
        
  <div class="_cycle_f0165c">
    <div>
      <canvas id="timer-cycle-test-17"></canvas>
    </div>
  </div>
        <div id="icon-play-pause-test-17" class="_icon-play-pause_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5 14.5L14.5 10L7.5 5.5V14.5ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"/>
</svg>
</div>
        <div id="icon-stop-test-17" class="_icon-stop_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M6 14H14V6H6V14ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"/>
</svg>
</div>
      </div>
    </div>
      "
`;

exports[`when set theme > should show light theme when set light 2`] = `
"
    <div class="_container_f0165c">
      <div>
        
  <div class="_timer_f0165c">
    <div>
      <canvas id="timer-test-18"></canvas>
    </div>
  </div>

        <div class="_cover_f0165c"></div>
        
  <div class="_cycle_f0165c">
    <div>
      <canvas id="timer-cycle-test-18"></canvas>
    </div>
  </div>
        <div id="icon-play-pause-test-18" class="_icon-play-pause_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5 14.5L14.5 10L7.5 5.5V14.5ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"></path>
</svg>
</div>
        <div id="icon-stop-test-18" class="_icon-stop_f0165c"><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M6 14H14V6H6V14ZM10 20C8.61667 20 7.31667 19.7375 6.1 19.2125C4.88333 18.6875 3.825 17.975 2.925 17.075C2.025 16.175 1.3125 15.1167 0.7875 13.9C0.2625 12.6833 0 11.3833 0 10C0 8.61667 0.2625 7.31667 0.7875 6.1C1.3125 4.88333 2.025 3.825 2.925 2.925C3.825 2.025 4.88333 1.3125 6.1 0.7875C7.31667 0.2625 8.61667 0 10 0C11.3833 0 12.6833 0.2625 13.9 0.7875C15.1167 1.3125 16.175 2.025 17.075 2.925C17.975 3.825 18.6875 4.88333 19.2125 6.1C19.7375 7.31667 20 8.61667 20 10C20 11.3833 19.7375 12.6833 19.2125 13.9C18.6875 15.1167 17.975 16.175 17.075 17.075C16.175 17.975 15.1167 18.6875 13.9 19.2125C12.6833 19.7375 11.3833 20 10 20ZM10 18C12.2333 18 14.125 17.225 15.675 15.675C17.225 14.125 18 12.2333 18 10C18 7.76667 17.225 5.875 15.675 4.325C14.125 2.775 12.2333 2 10 2C7.76667 2 5.875 2.775 4.325 4.325C2.775 5.875 2 7.76667 2 10C2 12.2333 2.775 14.125 4.325 15.675C5.875 17.225 7.76667 18 10 18Z" fill="#FEF7F7"></path>
</svg>
</div>
      </div>
    </div>
      "
`;
